packages/shell-api/src/shell-api.spec.ts

```diff
@@ -845,49 +849,49 @@ describe('ShellApi', function () {
 
         it('can get/set/list config keys', async function () {
           const value = { structuredData: 'value' };
-          expect(await config.set('somekey', value)).to.equal(
+          expect(await config.set('somekey' as any, value)).to.equal(
             'Setting "somekey" has been changed'
           );
-          expect(await config.get('somekey')).to.deep.equal(value);
+          expect(await config.get('somekey' as any)).to.deep.equal(value);
           expect((await toShellResult(config)).printable).to.deep.equal(
             new Map([['somekey', value]])
           );
-          expect(await config.reset('somekey')).to.deep.equal(
+          expect(await config.reset('somekey' as any)).to.deep.equal(
             'Setting "somekey" has been reset to its default value'
           );
-          expect(await config.get('somekey')).to.deep.equal('');
+          expect(await config.get('somekey' as any)).to.deep.equal('');
         });
 
         it('will fall back to defaults', async function () {
           expect(await config.get('displayBatchSize')).to.equal(20);
         });
 
         it('rejects setting unavailable config keys', async function () {
-          expect(await config.set('unavailable', 'value')).to.equal(
+          expect(await config.set('unavailable' as any, 'value')).to.equal(
             'Option "unavailable" is not available in this environment'
           );
         });
 
         it('rejects setting explicitly ignored config keys', async function () {
-          expect(await config.set('ignoreme', 'value')).to.equal(
+          expect(await config.set('ignoreme' as any, 'value')).to.equal(
             'Option "ignoreme" is not available in this environment'
           );
         });
 
         it('rejects setting explicitly invalid config values', async function () {
           store.badvalue = 1; // Make sure the config option exists
           validators.badvalue = (value) => `Bad value ${value} passed`;
-          expect(await config.set('badvalue', 'somevalue')).to.equal(
+          expect(await config.set('badvalue' as any, 'somevalue')).to.equal(
             'Cannot set option "badvalue": Bad value somevalue passed'
           );
         });
       });
 
       context('with a no-config evaluation listener', function () {
-        let config;
+        let config: ShellConfig;
 
         beforeEach(function () {
-          config = instanceState.context.config;
+          config = instanceState.shellApi.config;
```

Comment:
if all the previous assertions were running on instanceState.context, i expect either them to be wrong/fail or something i am not able to follow.

Reply:
Ah, these are the same objects, except that using `.shellApi` instead of `.context` makes this typesafe

`.context` is the global object in which shell commands are being executed, it more or less has to be typed as `any`, and properties from `.shellApi` are just copied over to it:

https://github.com/mongodb-js/mongosh/blob/1570f12d8dcdbf4668e2323edc124fd9c467e9f5/packages/shell-api/src/shell-instance-state.ts#L310-L311